### ISSUE TITLE
Bug: Valid Refresh Tokens despite user changing password

### ISSUE THREAD

**Issue by epicadk** [anitab-org/mentorship-backend#903]
**Describe the bug**

Refresh Tokens are still valid even after the user changes  passwords. 
**To Reproduce**

1. Login and save the refresh token you get. 
2. Change password 
3. Scroll down to Refresh endpoint and use the old Refresh token. 

**Expected behavior**

Refresh Tokens should not be valid after a user changes passwords. 

**Additional context**

This can be done by using the users hashed password as the secret for the refresh tokens.

**Comment by devkapilbansal** [comment]
I don't know whether it is the case. Will verify today. Thanks @epicadk for opening this

**Comment by isabelcosta** [comment]
@devkapilbansal if you can verify this, it would be amazing.

@epicadk can you please add more information to the ticket, for example how to reproduce this bug you found, so that whoever works on it, can have an example to follow. Also if you have any idea of a potential solution, even if its not the one being implemented, please put it in the alternatives :)

**Comment by epicadk** [comment]
> @devkapilbansal if you can verify this, it would be amazing.
> 
> @epicadk can you please add more information to the ticket, for example how to reproduce this bug you found, so that whoever works on it, can have an example to follow. Also if you have any idea of a potential solution, even if its not the one being implemented, please put it in the alternatives :)

@isabelcosta done . @devkapilbansal I have added steps to reproduce the issues please let me know if I should elaborate any further.

**Comment by devkapilbansal** [comment]
Valid Issue :heavy_check_mark: 

Thanks @epicadk to point out this security bug

Tested locally and able to get new access token using old refresh token that should not happen

**TLDR**

- Changing Password Screenshot
![change_password](https://user-images.githubusercontent.com/42766576/100550380-a71fcc80-329f-11eb-86b5-2a68593dc158.png)

- Successful Refresh
![successful_refresh](https://user-images.githubusercontent.com/42766576/100550389-b272f800-329f-11eb-9a9b-f50f8c5810b1.png)

Access token generated is also valid and can be used after refresh.

**Comment by devkapilbansal** [comment]
Also, I noticed that in the refresh endpoint it should mention refresh token instead of access token here

![refresher](https://user-images.githubusercontent.com/42766576/100550405-ca4a7c00-329f-11eb-8b32-e7a9c098bb98.png)

Therefore, opening an issue for this too

**Comment by devkapilbansal** [comment]
@isabelcosta @vj-codes @rpattath please label this issue

**Comment by isabelcosta** [comment]
thank you for such a thorough test and showing the output here 🙌 @devkapilbansal

**Comment by isabelcosta** [comment]
@devkapilbansal can you link up the issue here, in case you already created it?

**Comment by devkapilbansal** [comment]
> @devkapilbansal can you link up the issue here, in case you already created it?

The issue is #932

**Comment by tichnas** [comment]
@isabelcosta @devkapilbansal Can I please get assigned as no one is working on this issue?
Thanks

**Comment by devkapilbansal** [comment]
@tichnas consider asking it on zulip. You can work on this as soon as you are assigned to it

**Comment by gaurivn** [comment]
Assigned @tichnas

**Comment by tichnas** [comment]
Thanks a lot @gaurivn

**Comment by gaurivn** [comment]
You're welcome

**Comment by epicadk** [comment]
@isabelcosta an update working on the doc right now as discussed in the Mentorship system open session. Sorry it's taking so long.

**Comment by isabelcosta** [comment]
Thank you for the update! That's ok @epicadk :)
